streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordQueue.java

```diff
@@ -181,6 +182,11 @@ public Long headRecordOffset() {
         return headRecord == null ? null : headRecord.offset();
     }
 
+    @SuppressWarnings("OptionalAssignedToNull")
+    public Optional<Integer> headRecordLeaderEpoch() {
+        return headRecord == null ? null : headRecord.leaderEpoch();
```

Comment:
```suggestion
        return headRecord == null ? Optional.empty() : headRecord.leaderEpoch();
```

Reply:
Yeah we are distinguishing three cases here: `null` - queue empty. `Optional.empty()` - queue not empty, leader epoch not set on head record. otherwise, leader epoch. This is a bit confusing, but I think it's correct.

Reply:
Thanks. Time for an in-line comment.

Reply:
@alhsaeedi I think @mjsax suggested adding an inline comment explaining this. Could you please add it?